src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2052,103 +2022,93 @@ ostree_sysroot_deploy_tree (OstreeSysroot     *self,
                             GCancellable      *cancellable,
                             GError           **error)
 {
-  gboolean ret = FALSE;
-  gint new_deployserial;
-  glnx_unref_object OstreeDeployment *new_deployment = NULL;
-  glnx_unref_object OstreeDeployment *merge_deployment = NULL;
-  glnx_unref_object OstreeRepo *repo = NULL;
-  g_autoptr(GFile) osdeploydir = NULL;
-  g_autoptr(GFile) deployment_var = NULL;
-  glnx_fd_close int tree_boot_dfd = -1;
-  g_autofree char *tree_kernel_path = NULL;
-  g_autofree char *tree_initramfs_path = NULL;
-  glnx_fd_close int deployment_dfd = -1;
-  glnx_unref_object OstreeSePolicy *sepolicy = NULL;
-  g_autofree char *new_bootcsum = NULL;
-  glnx_unref_object OstreeBootconfigParser *bootconfig = NULL;
-
   g_return_val_if_fail (osname != NULL || self->booted_deployment != NULL, FALSE);
 
   if (osname == NULL)
     osname = ostree_deployment_get_osname (self->booted_deployment);
 
-  osdeploydir = ot_gfile_get_child_build_path (self->path, "ostree", "deploy", osname, NULL);
-  if (!g_file_query_exists (osdeploydir, NULL))
-    {
-      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
-                   "No OS named \"%s\" known", osname);
-      goto out;
-    }
-
-  deployment_var = g_file_get_child (osdeploydir, "var");
+  const char *osdeploypath = glnx_strjoina ("ostree/deploy/", osname);
+  glnx_fd_close int os_deploy_dfd = -1;
+  if (!glnx_opendirat (self->sysroot_fd, osdeploypath, TRUE, &os_deploy_dfd, error))
+    return FALSE;
 
+  glnx_unref_object OstreeRepo *repo = NULL;
   if (!ostree_sysroot_get_repo (self, &repo, cancellable, error))
-    goto out;
+    return FALSE;
 
+  glnx_unref_object OstreeDeployment *merge_deployment = NULL;
   if (provided_merge_deployment != NULL)
     merge_deployment = g_object_ref (provided_merge_deployment);
 
+  gint new_deployserial;
   if (!allocate_deployserial (self, osname, revision, &new_deployserial,
                               cancellable, error))
-    goto out;
+    return FALSE;
 
-  new_deployment = ostree_deployment_new (0, osname, revision, new_deployserial,
-                                          new_bootcsum, -1);
+  g_autofree char *new_bootcsum = NULL;
```

Comment:
Nit (since the goal here is to declare near the first use): there's no point in declaring `new_bootcsum` here if we can just pass `NULL` to `ostree_deployment_new`. Would make more sense to declare it down below before we call `checksum_from_kernel_src()`.